### PR TITLE
d/tests: replace broken `upgrade` test with simple `--help` call

### PR DESCRIPTION
**Pull request by paride**
The package had an autopkgtest that called the `upgrade` script.
That's not going to work as the `upgrade` script requires preparation
done via the `auto-upgrade-testing` command and that is based on a
configuration file (the test specification passed via --config).

In principle auto-upgrade-testing could itself be tested via an
autopkgtest, but that requires some non trivial work. For now let's
remove the broken autopkgtest and add a superficial `--help` call,
useful to spot major issues with the package (e.g. see d25cdf9d95).

### CODE REVIEW

**Comment by paride** [comment]
The diff is big because `upgrade_testing/data/upgrade` used to be a symlink to `debian/tests/upgrade`, but we don't want it to be. I think that's legacy of the first way `auto-upgrade-testing` was implemented.

**Comment by paride** [comment]
Once approved/merged I'll upload to Kinetic.

File: debian/changelog [hunk 1]
```diff
@@ -1,8 +1,10 @@
 auto-upgrade-testing (0.3.1) UNRELEASED; urgency=medium
 
   * debian/control: Add a dependency on python3-junitparser.
+  * d/tests: replace broken `upgrade` test with simple `--help` call
+  * d/gbp.conf: dch: don't merge commits by maintainer (multimaint = false)
 
- -- Brian Murray <brian@ubuntu.com>  Wed, 14 Sep 2022 17:01:26 -0700
+ -- Paride Legovini <paride@ubuntu.com>  Thu, 15 Sep 2022 13:09:13 +0200
```
Comment (murraybd) [inline debian/changelog:7]: I'm not fussed about having my name in the changelog but is there a reason you didn't use `dch -a`?

**Review by murraybd — APPROVED** [review]
I'm not worried about my name not appearing in the changelog but is there a reason you didn't use `dch -a`? When we have other team members it'd be good to acknowledge their work.

**Comment by paride** [comment]
> I'm not worried about my name not appearing in the changelog but is there a reason you didn't use `dch -a`? When we have other team members it'd be good to acknowledge their work.

I used gbp dch, which always changes the changelog trailer (which kind of makes sense, at least to update the timestamp). I can regenerate it in multimaint mode to give better attribution.

**Comment by paride** [comment]
@murraybd done, please have another look. Note that now this is a release changelog (targets kinetic instead of UNRELEASED), so it's ready to be uploaded.

**Pull request merged**